api/contrib/envoy/extensions/filters/network/generic_proxy/v3/route.proto

```diff
@@ -17,6 +17,32 @@ option (xds.annotations.v3.file_status).work_in_progress = true;
 
 // [#protodoc-title: Generic Proxy Route Configuration]
 
+message VirtualHost {
+  // The name of the virtual host.
+  string name = 1 [(validate.rules).string = {min_len: 1}];
+
+  // A list of hosts that will be matched to this virtual host. Wildcard hosts are supported in
+  // the suffix or prefix form.
+  //
+  // Host search order:
+  //  1. Exact names: ``www.foo.com``.
+  //  2. Suffix wildcards: ``*.foo.com`` or ``*-bar.foo.com``.
+  //  3. Prefix wildcards: ``foo.*`` or ``foo-*``.
+  //  4. Special wildcard ``*`` matching any host and will be the default virtual host.
+  //
+  // .. note::
+  //
+  //   The wildcard will not match the empty string.
+  //   e.g. ``*-bar.foo.com`` will match ``baz-bar.foo.com`` but not ``-bar.foo.com``.
+  //   The longest wildcards match first.
+  //   Only a single virtual host in the entire route configuration can match on ``*``. A domain
+  //   must be unique across all virtual hosts or the config will fail to load.
+  repeated string hosts = 2 [(validate.rules).repeated = {min_items: 1}];
```

Comment:
Why we didn't use the same name `domains` just like in the HCM virtualhost?

Reply:
The generic proxy stream provide a interface named `host()` the return value of `host()` will be used to match this `hosts` fields. So, I think maybe `hosts` is better here. 🤔 WDYT?

Reply:
if someday...we have generic proxy to instead of HCM :) , then I would say `domain` better, since that is more easy for the user to migrate to generic proxy.

Reply:
Although I think It' almost impossible because the HTTP is too complex orz. But OK, lets call it `domains` to keep consistency with HCM.

Reply:
Hmmmm, a little hesitation, the field's parent message is `VirtualHost`, think `hosts` would be better again. orz.

Reply:
fine for me, just want to raise the question ,ensure we thought it carefully.

Reply:
Then I will keep the name. Believe if some one wants to construct a HTTP proxy powered by generic proxy, the `hosts` name wouldn't be a problem in practice. 😄